command/init.go

```diff
@@ -516,6 +519,44 @@ func (c *InitCommand) getProviders(config *configs.Config, state *states.State,
 						),
 					))
 				}
+			case getproviders.ErrHostNoProviders:
+				switch {
+				case errorTy.Hostname == svchost.Hostname("github.com") && !errorTy.HasOtherVersion:
+					// If a user copies the URL of a GitHub repository into
+					// the source argument and removes the schema to make it
+					// provider-address-shaped then that's one way we can end up
+					// here. We'll use a specialized error message in anticipation
+					// of that mistake. We only do this if github.com isn't a
```

Comment:
FWIW saving this error for last is also a good decision because users could use "github.com" as the source host for locally-installed providers (really for whatever reason) and as long as the provider is properly placed on-disk it should be just fine to do so (weird, but fine).

I'm just reiterating this in part to remind myself :)